### ISSUE TITLE
pymc import issue - Mac OS 10.8

### ISSUE THREAD

**Issue by Flal** [pymc-devs/pymc3#312]
Hey guys,
I've installed PyMC using easy_install on the pymc egg file. Everything seemed to install correctly, but when I attempt to import pymc I get the following error:

> > > import pymc
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > >   File "/Library/Frameworks/Python.framework/Versions/7.3/lib/python2.7/site-packages/pymc-2.2-py2.7-macosx-10.7-intel.egg/pymc/**init**.py", line 25, in <module>
> > >     from .Container import *
> > >   File "/Library/Frameworks/Python.framework/Versions/7.3/lib/python2.7/site-packages/pymc-2.2-py2.7-macosx-10.7-intel.egg/pymc/Container.py", line 41, in <module>
> > >     from pymc.Container_values import LCValue, DCValue, ACValue, OCValue
> > > ImportError: dynamic module does not define init function (initContainer_values)

Any advice on how to fix this?

Thanks,

Alex

**Comment by fonnesbeck** [comment]
Hi Alex,

Is that the whole error message? Where did you get the .egg file? If you are on OS X 10.8, you can get reliable builds of PyMC along with the other key scientific programming packages in the [Scipy Superpack](http://bit.ly/scipy_superpack).

**Comment by Flal** [comment]
I installed the Scipy Superpack using:

$ sh install_superpack.sh

The install finished without any errors. However, I get the same error as before when I try to import pymc:

> > > import pymc
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > >   File "/Library/Frameworks/Python.framework/Versions/7.3/lib/python2.7/site-packages/pymc-2.2-py2.7-macosx-10.7-intel.egg/pymc/**init**.py", line 25, in <module>
> > >     from .Container import *
> > >   File "/Library/Frameworks/Python.framework/Versions/7.3/lib/python2.7/site-packages/pymc-2.2-py2.7-macosx-10.7-intel.egg/pymc/Container.py", line 41, in <module>
> > >     from pymc.Container_values import LCValue, DCValue, ACValue, OCValue
> > > ImportError: dynamic module does not define init function (initContainer_values)

Any thoughts? Am I missing a dependency?

**Comment by fonnesbeck** [comment]
OK, which Python are you using? I see that packages are being installed into `/Library/Frameworks/Python.framework/Versions/7.3/` rather than `/Library/Python/2.7`. The Superpack works best with Python 2.7.2 that ships with OS X.

**Comment by Flal** [comment]
I'm using Enthought's distributiuon (2.7.3). Has this been known to cause problems?

**Comment by fonnesbeck** [comment]
Well, Enthought already ships with most of the scientific packages, so you would have been better off simply building PyMC against the libraries that it already had. The error seems to suggest that there are issues with the compiled C extensions, which I cannot replicate here.

I'd suggest either using OSX's Python with the Superpack, or sticking with the packages in the Enthought distribtuions (or a third option is to use [Continuum's Anaconda](http://www.continuum.io/downloads)). Mixing the two is probably a mistake.

**Comment by patrick-roos** [comment]
I'm getting a similar error below on Ubuntu 13.04.
I installed the latest pymc using 
python setup.py config_fc --fcompiler gnu95 build
python setup.py install

Error: 

Enthought Canopy Python 2.7.3 | 32-bit [GCC 4.1.2 20080704 (Red Hat 4.1.2-52)] on linux2

> > > import pymc
> > > Warning: You are importing PyMC from inside its source tree.
> > > Traceback (most recent call last):
> > >  File "<stdin>", line 1, in <module>
> > >  File "pymc/**init**.py", line 26, in <module>
> > >    from .Container import *
> > > File "pymc/Container.py", line 41, in <module>
> > >    from .Container_values import LCValue, DCValue, ACValue, OCValue
> > > ImportError: No module named Container_values

**Comment by fonnesbeck** [comment]
This means there were problems building the C extensions. Can you confirm that you have a working gcc, and that it is being used by the setup? Can you post your entire build output so that I can track down where the build failed?

**Comment by patrick-roos** [comment]
I have a working gcc, I'm not totally sure how to check if it is being used by the setup.

Here's what I get from runnning

>  python setup.py install

:

> /usr/local/bin/Canopy/appdata/canopy-1.0.3.1262.rh5-x86/lib/python2.7/site-packages/numpy/distutils/system_info.py:564: UserWarning: Specified path /home/vagrant/src/master-env/lib is invalid.
>   warnings.warn('Specified path %s is invalid.' % d)
> /usr/local/bin/Canopy/appdata/canopy-1.0.3.1262.rh5-x86/lib/python2.7/site-packages/numpy/distutils/system_info.py:564: UserWarning: Specified path /home/vagrant/src/master-env/include is invalid.
>   warnings.warn('Specified path %s is invalid.' % d)
> /usr/local/bin/Canopy/appdata/canopy-1.0.3.1262.rh5-x86/lib/python2.7/site-packages/numpy/distutils/system_info.py:1408: UserWarning: 
>     Atlas (http://math-atlas.sourceforge.net/) libraries not found.
>     Directories to search for the libraries can be specified in the
>     numpy/distutils/site.cfg file (section [atlas]) or by setting
>     the ATLAS environment variable.
>   warnings.warn(AtlasNotFoundError.**doc**)
> /usr/local/bin/Canopy/appdata/canopy-1.0.3.1262.rh5-x86/lib/python2.7/site-packages/numpy/distutils/system_info.py:1419: UserWarning: 
>     Lapack (http://www.netlib.org/lapack/) libraries not found.
>     Directories to search for the libraries can be specified in the
>     numpy/distutils/site.cfg file (section [lapack]) or by setting
>     the LAPACK environment variable.
>   warnings.warn(LapackNotFoundError.**doc**)
> /usr/local/bin/Canopy/appdata/canopy-1.0.3.1262.rh5-x86/lib/python2.7/site-packages/numpy/distutils/system_info.py:1422: UserWarning: 
>     Lapack (http://www.netlib.org/lapack/) sources not found.
>     Directories to search for the sources can be specified in the
>     numpy/distutils/site.cfg file (section [lapack_src]) or by setting
>     the LAPACK_SRC environment variable.
>   warnings.warn(LapackSrcNotFoundError.**doc**)
> /usr/local/bin/Canopy/appdata/canopy-1.0.3.1262.rh5-x86/lib/python2.7/site-packages/numpy/distutils/system_info.py:495: UserWarning: 
>     Lapack (http://www.netlib.org/lapack/) libraries not found.
>     Directories to search for the libraries can be specified in the
>     numpy/distutils/site.cfg file (section [lapack]) or by setting
>     the LAPACK environment variable.
>   warnings.warn(self.notfounderror.**doc**)
> No optimized BLAS or Lapack libraries found, building from source. This may take a while...
> No optimized BLAS or Lapack libraries found, building from source. This may take a while...
> running install
> running build
> running config_cc
> unifing config_cc, config, build_clib, build_ext, build commands --compiler options
> running config_fc
> unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
> running build_src
> build_src
> building extension "pymc.flib" sources
> f2py options: []
>   adding 'build/src.linux-i686-2.7/fortranobject.c' to sources.
>   adding 'build/src.linux-i686-2.7' to include_dirs.
>   adding 'build/src.linux-i686-2.7/pymc/flib-f2pywrappers.f' to sources.
> building extension "pymc.LazyFunction" sources
> building extension "pymc.Container_values" sources
> building extension "pymc.gp.linalg_utils" sources
> f2py options: []
>   adding 'build/src.linux-i686-2.7/fortranobject.c' to sources.
>   adding 'build/src.linux-i686-2.7' to include_dirs.
>   adding 'build/src.linux-i686-2.7/pymc/gp/linalg_utils-f2pywrappers.f' to sources.
> building extension "pymc.gp.incomplete_chol" sources
> f2py options: []
>   adding 'build/src.linux-i686-2.7/fortranobject.c' to sources.
>   adding 'build/src.linux-i686-2.7' to include_dirs.
>   adding 'build/src.linux-i686-2.7/pymc/gp/incomplete_chol-f2pywrappers.f' to sources.
> building extension "pymc.gp.cov_funs.isotropic_cov_funs" sources
> f2py options: []
>   adding 'build/src.linux-i686-2.7/fortranobject.c' to sources.
>   adding 'build/src.linux-i686-2.7' to include_dirs.
>   adding 'build/src.linux-i686-2.7/pymc/gp/cov_funs/isotropic_cov_funs-f2pywrappers.f' to sources.
> building extension "pymc.gp.cov_funs.distances" sources
> f2py options: []
>   adding 'build/src.linux-i686-2.7/fortranobject.c' to sources.
>   adding 'build/src.linux-i686-2.7' to include_dirs.
> build_src: building npy-pkg config files
> running build_py
> running build_ext
> customize UnixCCompiler
> customize UnixCCompiler using build_ext
> customize Gnu95FCompiler
> Found executable /usr/bin/gfortran
> customize Gnu95FCompiler
> customize Gnu95FCompiler using build_ext
> running scons
> running install_lib
> copying build/lib.linux-i686-2.7/pymc/Container_values.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc
> copying build/lib.linux-i686-2.7/pymc/LazyFunction.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc
> copying build/lib.linux-i686-2.7/pymc/flib.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc
> copying build/lib.linux-i686-2.7/pymc/gp/linalg_utils.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc/gp
> copying build/lib.linux-i686-2.7/pymc/gp/cov_funs/isotropic_cov_funs.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc/gp/cov_funs
> copying build/lib.linux-i686-2.7/pymc/gp/cov_funs/distances.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc/gp/cov_funs
> copying build/lib.linux-i686-2.7/pymc/gp/incomplete_chol.so -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc/gp
> copying build/lib.linux-i686-2.7/pymc/tests/test_LazyFunction.py -> /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc/tests
> byte-compiling /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc/tests/test_LazyFunction.py to test_LazyFunction.pyc
> running install_egg_info
> Removing /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc-2.2-py2.7.egg-info
> Writing /home/roos/Enthought/Canopy_32bit/User/lib/python2.7/site-packages/pymc-2.2-py2.7.egg-info
> running install_clib
> customize UnixCCompiler

Thanks!

**Comment by patrick-roos** [comment]
Update: I installed gcc 4.8 (replacing 4.7) and also reinstalled numpy. 

Then ran the setup.py install again and that fixed it. 

 Thanks for your work!

**Comment by fonnesbeck** [comment]
Excellent news.

**Comment by Flal** [comment]
Update: it seems that the problems with the Enthought package is that the libraries it includes in its distro don't always use the architecture required by PyMC. I tried customizing my libraries but just ended up making a mess of everything (I'm still fairly new to this so someone with more experience may have better luck).

I finally wiped all versions of Python from my computer and installed a fresh version of 2.7.5. When I ran the Scipy Superpack script, everything installed properly and PyMC works like a charm.

Thanks for the help!
